.github/workflows/build.yml

```diff
@@ -7,36 +7,68 @@ on:
     branches:
       - main
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref_name }}
+  cancel-in-progress: true
+
 jobs:
-  build:
-    name: Build Firmware
+  configure:
+    name: Verify ${{ matrix.appliance.name }} Configuration with ${{ matrix.config_file }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        appliance:
+          - name: electric_tank_water_heater
+            shorthand: etwh
+          - name: heatpump_water_heater
+            shorthand: hpwh
+          - name: hvac
+            shorthand: hvac
+          - name: tankless_water_heater
+            shorthand: tlwh
+        config_file: [example-esp32.yaml, example-esp8266.yaml, example-local.yaml]
+      fail-fast: false
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v4
+      - uses: actions/checkout@v4.1.1
+      - uses: actions/setup-python@v4.7.1
         with:
           python-version: "3.x"
       - name: Install ESPHome
-        run: pip install --upgrade esphome
-      - name: Create Stub Secrets File
-        run: |
-          echo 'wifi_ssid: "STUB_SSID"' >> secrets.yaml
-          echo 'wifi_password: "STUB_PASSWORD"' >> secrets.yaml
-      - name: Verify All Configurations
-        run: |
-          esphome -s external_components_source components config \
-          econet_electric_tank_water_heater.yaml \
-          econet_heatpump_water_heater.yaml  \
-          econet_hvac.yaml \
-          econet_tankless_water_heater.yaml \
-          example-local.yaml
-      - name: Compile Firmware
-        run: |
-          esphome -s external_components_source components compile \
-          econet_electric_tank_water_heater.yaml \
-          econet_heatpump_water_heater.yaml  \
-          econet_hvac.yaml \
-          econet_tankless_water_heater.yaml
-    concurrency:
-      group: ${{ github.workflow }}-${{ github.head_ref || github.ref_name }}
-      cancel-in-progress: true
+        run: pip install esphome
+      - name: Update manifest
+        uses: mikefarah/yq@v4.35.2
+        with:
+          cmd: >
+            yq -P -i '.packages.econet = "econet_${{ matrix.appliance.name }}.yaml"
+            | .packages.econet tag = "!include"
+            | .wifi.ssid = "STUB_SSID"
+            | .wifi.ssid tag = ""
+            | .wifi.password = "STUB_PASSWORD"
+            | .wifi.password tag = ""'
+            ./${{ matrix.config_file }}
+      - name: Validate configuration
+        run: esphome -s external_components_source components config ./${{ matrix.config_file }}
+
+  build:
+    name: Build ${{ matrix.appliance.name }} Firmware
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        appliance:
```

Comment:
Similar to the other PR, is it possible to define this map once in this file? Is it possible to define it once for all workflows?

Reply:
I can't find a way - we can try to refactor in a future update.